cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/ConceptPass.kt

```diff
@@ -46,9 +48,16 @@ abstract class ConceptPass(ctx: TranslationContext) : TranslationUnitPass(ctx) {
     override fun accept(tu: TranslationUnitDeclaration) {
         ctx.currentComponent = tu.component
         walker = SubgraphWalker.ScopedWalker(ctx.scopeManager)
+        walker.strategy = Strategy::EOG_FORWARD
         walker.registerHandler { node -> handleNode(node, tu) }
 
-        walker.iterate(tu)
+        // Gather all resolution EOG starters; and make sure they really do not have a
+        // predecessor, otherwise we might analyze a node multiple times
+        val nodes = tu.allEOGStarters.filter { it.prevEOGEdges.isEmpty() }
```

Comment:
Don't we have similar code already somewhere?

Reply:
In theory there is the EOGStarterPass, but if I change it to one, I somehow get incorrect results when retrieving  the current TU :(